pkg/models/datasource_cache.go

```diff
@@ -86,8 +156,9 @@ func (ds *DataSource) GetHttpTransport() (*dataSourceTransport, error) {
 	}
 
 	dsTransport := &dataSourceTransport{
-		headers:   customHeaders,
-		transport: transport,
+		headers:        customHeaders,
+		transport:      transport,
```

Comment:
Nit. Would construct the roundtrippers here and use that as transport, e.g. 

```go 
type RoundTripperFunc func(req *http.Request) (*http.Response, error)

func customHeadersRoundtrip(headers map[string]string, next http.RoundTripper) RoundTripperFunc { ... } 

roundTripper := customHeadersRoundtrip(customHeaders, instrumentRoundtrip(d.datasourceName, transport))

dsTransport := &dataSourceTransport{
  transport: roundTripper,
}
```

This way easier to add additional roundtrippers (auth middleware in the pipeline etc :) )

Reply:
I picked this solution to make it as simple as possible since I know you are gonna be working on it. Is it fine if you address the additional roundtrippers in your PR?

Reply:
Sure!